daemon/cluster/cluster.go

```diff
@@ -235,7 +235,7 @@ func (c *Cluster) reconnectOnFailure(n *node) {
 	}
 }
 
-func (c *Cluster) startNewNode(forceNewCluster bool, localAddr, remoteAddr, listenAddr, advertiseAddr, joinAddr, joinToken string) (*node, error) {
+func (c *Cluster) startNewNode(forceNewCluster bool, localAddr, remoteAddr, listenAddr, advertiseAddr, joinAddr, joinToken string, availability types.NodeAvailability) (*node, error) {
```

Comment:
Thanks @stevvooe. The pull request has been updated with `localAddr`, `listenAddr`, and `advertiseAddr` into one consolidated `nodeAddr`. Please let me know if there are any issues.